bazel/linux/defs.bzl

```diff
@@ -261,37 +329,68 @@ def kernel_module(*args, **kwargs):
         kwargs["srcs"] = native.glob(include = include, exclude = BUILD_LEFTOVERS, allow_empty = False)
 
-    module = kwargs.get("module", kwargs["name"])
-    if not module.endswith(".ko"):
-        module = module + ".ko"
-    kwargs["module"] = module
+    if "module" in kwargs:
+        module = kwargs["module"]
+        kwargs.pop("module")
+    else:
+        module = kwargs["name"]
 
-    if "kernels" not in kwargs:
-        kwargs["kernel"] = _normalize_kernel(kwargs["kernel"])
-        return kernel_module_rule(*args, **kwargs)
+    kwargs["modules"] = [module]
 
-    kernels = kwargs["kernels"]
-    kwargs.pop("kernels")
+    if "make_format_str" not in kwargs:
+        kwargs["make_format_str"] = "-C $PWD/{kernel_build_dir} M=$PWD/{src_dir} {modules}"
 
-    targets = []
     original = kwargs["name"]
-    for kernel in kernels:
-        kernel = _normalize_kernel(kernel)
-        rename = kernel[1:] + "/" + module
-        name = kernel[1:] + "-" + original
-        targets.append(":" + name)
+    targets = _kernel_module_targets(*args, **kwargs)
 
-        kwargs["name"] = name
-        kwargs["kernel"] = kernel
-        kwargs["rename"] = rename
-        kernel_module_rule(*args, **kwargs)
+    # This creates a target with the name chosen by the user that builds all the kernel modules at once.
+    # Without this, the user can only build :all, or the specific module for a specific kernel.
+    return native.filegroup(name = original, srcs = targets)
+
+def nv_driver(*args, **kwargs):
+    """Convenience wrapper around kernel_modules_rule.
+
+    Use this wrpaper for building the NVidia driver modules.
+
+    The parameters passed to nv_driver are just passed to nv_driver_rule, except for
+    what is listed below.
+
+    Args:
+      srcs: list of labels, specifying the source files that constitute the kernel module.
+            If not specified, nv_driver will provide a reasonable default including all
+            files that are typically part of a kernel module (i.e., the specified makefile
+            and all .c and .h files belonging to the package where the kernel_module rule
+            has been instantiated, see https://docs.bazel.build/versions/master/be/functions.html#glob).
+      modules: list of strings, naming the output modules. Mandatory. Also, it normalizes the
+            names ensuring they have a '.ko' suffix.
+      makefile: string, name of the makefile to build the driver. If not specified, nv_driver
+            assumes it is just called Makefile.
+      kernel: a label, indicating the kernel_tree to build the module against. nv_driver ensures
+            the label starts with an '@', as per bazel convention.
+      kernels: list of kernel (same as above). kernel_module will instantiate multiple
+            kernel_module_rule, one per kernel, and ensure they all build in parallel.
+    """
+
+    if "makefile" not in kwargs:
+        kwargs["makefile"] = "Makefile"
+
+    if "srcs" not in kwargs:
```

Comment:
I was about to suggest to move this "srcs" in a separate function, but then I realized that 1) the list is different from kernel_module, but the rest of the mechanic is pretty much the same as `kernel_module`?

Not sure this is overkill, feel free to do it or not. But suggestion:

a) add a `_kernel_default_srcs(extra_includes = [], extra_excludes = [])` function that pretty much `return native.glob ... include + extra_includes, exclude = BUILD_LEFTOVERS + extra_excludes, ...)`

b) use `_kernel_default_srcs()` both here and above.

Reply:
also nit: just remembered that skylark supports setdefault, so you could:

```
kwargs.setdefault("makefile", "Makefile")
kwargs.setdefault("srcs", _kernel_default_srcs(includes = kwargs["makefile"]))
kwargs.setdefault("make_format_str", ...)
```
at your taste.